astropy/table/table.py

```diff
@@ -109,6 +109,8 @@ def __getitem__(self, item):
             return OrderedDict.__getitem__(self, item)
         elif isinstance(item, (int, np.integer)):
             return self.values()[item]
+        elif (isinstance(item, np.ndarray) and item.shape == () and item.dtype.kind == 'i'):
+            return self.values()[item.item()]
```

Comment:
So it seems that you found the same bug while indexing row values ? (This is why tests are essential !)